src/renderers/dom/shared/__tests__/ReactDOMServerIntegration-test.js

```diff
@@ -1022,6 +1024,552 @@ describe('ReactDOMServerIntegration', () => {
     });
   });
 
+  describe('form controls', function() {
+    describe('inputs', function() {
+      itRenders('an input with a value and an onChange', async render => {
+        const e = await render(<input value="foo" onChange={() => {}} />);
+        expect(e.value).toBe('foo');
+      });
+
+      itRenders('an input with a value and readOnly', async render => {
+        const e = await render(<input value="foo" readOnly={true} />);
+        expect(e.value).toBe('foo');
+      });
+
+      itRenders(
+        'an input with a value and no onChange/readOnly',
+        async render => {
+          // this configuration should raise a dev warning that value without
```

Comment:
I agree; I'm genuinely not certain if it's a good idea to have in SSR or not. If not, we can just change these tests pretty easily, though.